docs/advanced-features/using-mdx/mdx-bundler.md

```diff
@@ -0,0 +1,232 @@
+---
+description: Learn how to use mdx-bundler in your Next.js project.
+---
+
+# `mdx-bundler`
+
+The `mdx-bundler` package does **not** require a global configuration, instead it bundles the dependencies of your MDX files allowing you to directly import components in the MDX file. `mdx-bundler` also supports soucing data from anywhere, including local files, external APIs or databases.
+
+Note that it uses [esbuild](https://esbuild.github.io) under the hood, meaning this will need to be installed as a dependency.
+
+Also note that one of `mdx-bundler`'s dependencies requires a working node-gyp setup to be able to install correctly.
+
+## Setup `mdx-bundler` in Next.js
+
+The following shows how to setup `mdx-bundler` for use with a blog.
+
+1. Install the required packages:
+
+   ```terminal
+     npm i mdx-bundler esbuild gray-matter
+
+     #or
+
+     yarn add mdx-bundler esbuild gray-matter
+   ```
+
+2. Create an `mdx.js` file. The name is arbitrary and the file can be placed anywhere within your projects file structure. In this file you will setup `mdx-bundler` and create some utility functions that will get your MDX files content, frontmatter, and as a nice touch, provide a previous and next path.
+
+3. The below example assumes you have a directory within `/pages` called `/posts`. Begin by creating a function that will read the file contents from a specified path. In this case, a path to your blog posts:
+
+   ```js
+   import fs from 'fs'
+   import path from 'path'
+   import matter from 'gray-matter'
```

Comment:
btw the `unified` ecosystem also has a utility to extract and parse frontmatter: https://github.com/vfile/vfile-matter